sensing/preprocessor/gnss/gnss_poser/src/gnss_poser_core.cpp

```diff
@@ -13,76 +13,57 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-#include "gnss_poser/gnss_poser_core.h"
+#include "gnss_poser/gnss_poser_core.hpp"
+#include <rclcpp_components/register_node_macro.hpp>
 
 #include <algorithm>
 #include <string>
 
-#include <boost/circular_buffer.hpp>
-
-#include <ros/ros.h>
-
-#include <tf2/transform_datatypes.h>
-#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
-#include <tf2_ros/transform_broadcaster.h>
-
-#include <geometry_msgs/PoseStamped.h>
-#include <geometry_msgs/PoseWithCovarianceStamped.h>
-#include <sensor_msgs/NavSatFix.h>
-#include <std_msgs/Bool.h>
-#include <ublox_msgs/NavPVT.h>
-
-#include "gnss_poser/convert.h"
-
-namespace GNSSPoser
-{
-GNSSPoser::GNSSPoser(ros::NodeHandle nh, ros::NodeHandle private_nh)
-: nh_(nh),
-  private_nh_(private_nh),
+GNSSPoser::GNSSPoser(const rclcpp::NodeOptions & node_options)
+: rclcpp::Node("gnss_poser", node_options),
   tf2_listener_(tf2_buffer_),
-  coordinate_system_(CoordinateSystem::MGRS),
-  base_frame_("base_link"),
-  gnss_frame_("gnss"),
-  gnss_base_frame_("gnss_base_link"),
-  map_frame_("map"),
-  use_ublox_receiver_(false),
-  plane_zone_(9)
+  tf2_broadcaster_(this),
+  clock_(RCL_ROS_TIME)
 {
-  int coordinate_system = static_cast<int>(coordinate_system_);
-  private_nh_.getParam("coordinate_system", coordinate_system);
+  int coordinate_system =
+    declare_parameter("coordinate_system", static_cast<int>(CoordinateSystem::MGRS));
   coordinate_system_ = static_cast<CoordinateSystem>(coordinate_system);
 
-  private_nh_.getParam("base_frame", base_frame_);
-  private_nh_.getParam("gnss_frame", gnss_frame_);
-  private_nh_.getParam("gnss_base_frame", gnss_base_frame_);
-  private_nh_.getParam("map_frame", map_frame_);
-  private_nh_.getParam("use_ublox_receiver", use_ublox_receiver_);
-  private_nh_.getParam("plane_zone", plane_zone_);
+  base_frame_ = declare_parameter("base_frame", "base_link");
```

Comment:
I looked at `declare_parameter` yesterday myself in porting `mpc_follower`. Haven't gotten the whole code to compile so I could test this myself but it looks ok. If you want to support only the use-case to set values initially through e.g. the launch file, then, however, we should  declare the members like `gnss_frame` as `const` and move the initialization to the initializer list:

```c++
GNSSPoser::GNSSPoser(const rclcpp::NodeOptions & node_options) :
...
gnss_frame(declare_parameter("gnss_frame", "gnss"))
```

But again, this is a suggestion that goes slightly beyond porting alone

Reply:
These are good suggestions I will add them, though they are going slightly beyond a porting. I think the parameters are unlikely to change during runtime so these are innocuous changes.